src/main/java/cn/nukkit/level/provider/leveldb/serializer/LevelDBDataSerializer.java

```diff
@@ -151,11 +152,12 @@ private void loadData(LevelData data, Path levelDatPath) throws IOException {
         Map<String, Object> tagMap = tag.parseValue();
         GameRuleRegistry.get().getRules().forEach(rule -> {
             Object value = tagMap.get(rule.getName().toLowerCase());
-            if (value instanceof Byte) {
-                data.getGameRules().put((GameRule<Boolean>) rule, (byte) value != 0);
-            } else if (value instanceof Integer) {
+            Class type = rule.getValueClass();
```

Comment:
I don't get these lines, why modify?

Reply:
Because the way the tag was being read converted byte to ints, so Boolean tags were being parsed as Integer tags and then getting class cast exceptions.